platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfiguration.java

```diff
@@ -0,0 +1,216 @@
+ *                                   used
+ * @param busyFractionMetricEnabled  whether the busy fraction metric should be enabled, if null then false is used
+ * @param flushingEnabled            whether flushing is enabled, if null then false is used
+ * @param squelchingEnabled          whether squelching is enabled, if null then false is used
+ */
+public record TaskSchedulerConfiguration(
+        @Nullable TaskSchedulerType type,
+        @Nullable Long unhandledTaskCapacity,
+        @Nullable Boolean unhandledTaskMetricEnabled,
+        @Nullable Boolean busyFractionMetricEnabled,
+        @Nullable Boolean flushingEnabled,
+        @Nullable Boolean squelchingEnabled) {
+
+    /**
+     * Parse a string representation of a task scheduler configuration.
+     * <p>
+     * Syntax is as follows:
+     * <ul>
+     *     <li>
+     *         Zero or one values from the {@link TaskSchedulerType} enum, specifies the type of the task scheduler.
+     *         If not present then the default is used.
+     *     </li>
+     *     <li>
+     *         Zero or one string of the form "CAPACITY(1234)", specifies the maximum number of unhandled tasks.
+     *     </li>
+     *     <li>
+     *         Zero or more values from the {@link TaskSchedulerConfigOption} enum, specifies the configuration options.
+     *         Sets a boolean configuration option to true if the value is present, and false if the value is prefixed
+     *         with a "!". If not present then the default is used.
+     *     </li>
+     * </ul>
+     * Example: "SEQUENTIAL CAPACITY(500) !FLUSHABLE UNHANDLED_TASK_METRIC"
+     * <p>
+     * Note that default values are not specified within this class. Default values are the responsibility of the
+     * {@link TaskSchedulerBuilder} class.
+     *
+     * @param string the string to parse
+     * @return the parsed configuration
+     */
+    @NonNull
+    public static TaskSchedulerConfiguration parse(@NonNull final String string) {
+        TaskSchedulerType type = null;
+        Long unhandledTaskCapacity = null;
+        Boolean unhandledTaskMetricEnabled = null;
+        Boolean busyFractionMetricEnabled = null;
+        Boolean flushingEnabled = null;
+        Boolean squelchingEnabled = null;
+
+        final String[] parts = string.split(" ");
+        for (final String part : parts) {
+            final String strippedPart = part.strip();
+            if (strippedPart.isEmpty()) {
+                continue;
+            }
+
+            final TaskSchedulerType parsedType = tryToParseTaskSchedulerType(strippedPart);
+            if (parsedType != null) {
+                if (type != null) {
+                    throw new IllegalArgumentException("Multiple task scheduler types specified: " + string);
+                }
+                type = parsedType;
+                continue;
+            }
+
+            final Long parsedCapacity = tryToParseCapacity(strippedPart);
```

Comment:
Minor: perhaps using a regular expression to parse something like `"(\!)?(\w+)(?:\((\d+)\))?"` and implementing a `static TaskSchedulerConfigOption.valueOf(String)` to parse the special CAPACITY and the rest of the enum constants (using the Enum.valueOf), and then running a simple `switch (valueOf(matcher.group(2)))` here could result in a more robust parser by avoiding low-level strings parsing in the try*() methods.

Reply:
My personal preference for this parsing is to keep the initial implementation super simple, and to consider more complex/complete parsing solutions if we find this to be an area where we have growing pain. IMO, even if regex results in more robust code, I think manual string parsing is a lot simpler simpler given the size of the grammar we need to parse.

Reply:
Sounds fair.